src/k8s/pkg/utils/cidr.go

```diff
@@ -37,3 +60,30 @@ func GetKubernetesServiceIPsFromServiceCIDRs(serviceCIDR string) ([]net.IP, erro
 	}
 	return firstIPs, nil
 }
+
+// ParseAddressString parses an address string and returns a canonical network address.
+func ParseAddressString(address string, port int64) (string, error) {
+	host, hostPort, err := net.SplitHostPort(address)
+	if err == nil {
+		address = host
+		port, err = strconv.ParseInt(hostPort, 10, 64)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse the port from %q: %w", hostPort, err)
+		}
+	}
+
+	if address == "" {
+		address = util.NetworkInterfaceAddress()
+	}
+
+	if _, ipNet, err := net.ParseCIDR(address); err == nil {
+		matchingIP, err := FindMatchingNodeAddress(ipNet)
```

Comment:
we can inline this function, or perhaps keep it as private `findMatchingNodeAddress`